x-pack/test/api_integration/apis/ml/annotations/update_annotations.ts

```diff
@@ -0,0 +1,171 @@
+        await ml.api.indexAnnotation(annotationToIndex);
+      }
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should correctly update annotation by id', async () => {
+      const annotationsForJob = await ml.api.getAnnotations(jobIds[0]);
+      expect(annotationsForJob).to.have.length(1);
+
+      const originalAnnotation = annotationsForJob[0];
+      const annotationUpdateRequestBody = {
+        timestamp: Date.now(),
+        end_timestamp: Date.now(),
+        annotation: 'Updated annotation #0',
+        job_id: originalAnnotation._source.job_id,
+        type: ANNOTATION_TYPE.ANNOTATION,
+        event: 'model_change',
+        detector_index: 2,
+        partition_field_name: 'airline',
+        partition_field_value: 'ANA',
+        _id: originalAnnotation._id,
+      };
+      const { body } = await supertest
+        .put('/api/ml/annotations/index')
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(annotationUpdateRequestBody)
+        .expect(200);
+
+      expect(body._id).to.eql(originalAnnotation._id);
+      expect(body.result).to.eql('updated');
+
+      const updatedAnnotation = await ml.api.getAnnotationById(originalAnnotation._id);
+      expect(updatedAnnotation?.annotation).to.eql(annotationUpdateRequestBody.annotation);
+      expect(updatedAnnotation?.detector_index).to.eql(annotationUpdateRequestBody.detector_index);
+      expect(updatedAnnotation?.event).to.eql(annotationUpdateRequestBody.event);
+    });
+
+    it('should correctly update annotation for user with viewer permission', async () => {
+      const annotationsForJob = await ml.api.getAnnotations(jobIds[1]);
+      expect(annotationsForJob).to.have.length(1);
+
+      const originalAnnotation = annotationsForJob[0];
+      const annotationUpdateRequestBody = {
+        timestamp: Date.now(),
+        end_timestamp: Date.now(),
+        annotation: 'Updated annotation #1',
+        job_id: originalAnnotation._source.job_id,
+        type: ANNOTATION_TYPE.ANNOTATION,
+        event: 'model_change',
+        detector_index: 2,
+        _id: originalAnnotation._id,
+      };
+      const { body } = await supertest
+        .put('/api/ml/annotations/index')
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(annotationUpdateRequestBody)
+        .expect(200);
+
+      expect(body._id).to.eql(originalAnnotation._id);
+      expect(body.result).to.eql('updated');
+
+      const updatedAnnotation = await ml.api.getAnnotationById(originalAnnotation._id);
+      expect(updatedAnnotation?.annotation).to.eql(annotationUpdateRequestBody.annotation);
+      expect(updatedAnnotation?.detector_index).to.eql(annotationUpdateRequestBody.detector_index);
+      expect(updatedAnnotation?.event).to.eql(annotationUpdateRequestBody.event);
```

Comment:
If add partition details to the update, we should validate them here.

Reply:
Updated in https://github.com/elastic/kibana/pull/73068/commits/f45a447e78d38637d6649fcb92e6d3235b13c906